spec/models/user_spec.rb

```diff
@@ -2,23 +2,36 @@
 
 describe User, type: :model do
   describe "validations" do
-
-
-  it {should validate_presence_of(:name)}
-  it {should validate_presence_of(:address)}
-  it {should validate_presence_of(:city)}
-  it {should validate_presence_of(:state)}
-  it {should validate_presence_of(:zip)}
-  it {should validate_presence_of(:email)}
-  it {should validate_presence_of(:password)}
-
-
-
-
-
+    it {should validate_presence_of(:name)}
+    it {should validate_presence_of(:address)}
+    it {should validate_presence_of(:city)}
+    it {should validate_presence_of(:state)}
+    it {should validate_presence_of(:zip)}
+    it {should validate_presence_of(:email)}
+    it {should validate_presence_of(:password)}
   end
 
-  describe "Relationships" do
+  describe "relationships" do
     it {should have_many :orders}
   end
+
+  describe "roles" do
```

Comment:
Added these roles tests for role creation

Reply:
Nice work here